pkg/controllers/jobset_controller.go

```diff
@@ -207,18 +210,18 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 
 		// Create headless service if specified for this job.
 		if rjob.Network.EnableDNSHostnames != nil && *rjob.Network.EnableDNSHostnames {
-			klog.Infof("creating headless service: %s", job.Name)
+			log.Info("creating headless service", "service", job.Name)
 			if err := r.createHeadlessSvcIfNotExist(ctx, js, job); err != nil {
-				klog.Errorf("error creating headless service: %v", err)
+				log.Error(err, "error creating headless service", "service", job.Name)
 				return err
 			}
 		}
 
 		// Create the job.
 		// TODO: Deal with the case where the job exists but is not owned by the jobset.
-		klog.Infof("creating job %s", job.Name)
+		log.Info("creating job", "job", job.Name)
```

Comment:
Lets use v(2) for the logs that we want to show in a production run, and something higher for debugging